pkg/attribute/protoBag.go

```diff
@@ -50,9 +64,56 @@ func (pb *ProtoBag) Get(name string) (interface{}, bool) {
 	index, ok := pb.getIndex(name)
 	if !ok {
 		// the string is not in the dictionary, and hence the attribute is not in the proto either
+		pb.trackReference(name, mixerpb.ABSENCE)
 		return nil, false
 	}
 
+	result, ok := pb.internalGet(name, index)
+	if !ok {
+		// the named attribute was not present
+		pb.trackReference(name, mixerpb.ABSENCE)
+		return nil, false
+	}
+
+	pb.trackReference(name, mixerpb.EXACT)
+	return result, ok
+}
+
+// Get the set of attributes that have been referenced through this bag.
+func (pb *ProtoBag) GetReferencedAttributes() mixerpb.ReferencedAttributes {
+	output := mixerpb.ReferencedAttributes{}
+
+	ds := newDictState(nil)
+
+	output.AttributeMatches = make([]mixerpb.ReferencedAttributes_AttributeMatch, len(pb.referencedAttrs))
```

Comment:
any reason not use use `make([]..., 0, len(pb.referencedAttrs))` and `append`s (instead of `i`) ?